de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/browser/EclipseBrowserDialog.java

```diff
@@ -16,6 +16,7 @@
 
     public EclipseBrowserDialog(Shell shell) {
```

Comment:
The CTOR is really bad. If you look at the SWT framework and Eclipse stuff you will notice that the shell is always named parent for a good reason.

This class however expects a concrete shell which will be further mutated during the application lifetime which is really bad because the class is not the owner/creator of the shell.